sys/net/gnrc/application_layer/tftp/gnrc_tftp.c

```diff
@@ -233,7 +233,7 @@ static tftp_state _tftp_send_error(tftp_context_t *ctxt, gnrc_pktsnip_t *buf, tf
 static tftp_state _tftp_send(gnrc_pktsnip_t *buf, tftp_context_t *ctxt, size_t len);
 
 /* decode the default TFTP start packet */
-static int _tftp_decode_start(tftp_context_t *ctxt, uint8_t *buf, gnrc_pktsnip_t *outbuf);
+static int _tftp_decode_start(tftp_context_t *ctxt, uint8_t *buf, size_t len, gnrc_pktsnip_t *outbuf);
```

Comment:
I would rather change the function to

```
static int _tftp_decode_start(tftp_context_t *ctxt, uint8_t *buf,  gnrc_pktsnip_t *inpkt, gnrc_pktsnip_t *outbuf);
```

and in that function have

```
size_t len = inpkt->len
uint8_t *buf = inpkt->buf
```

what do you think?

Reply:
or instead of the latter cast directly

```
tftp_header_t *hdr = (tftp_header_t *)inpkt->buf;
```

Reply:
Changed the function parameter to contain a single input `gnrc_pktsnip_t*` instead of passing a buffer and length information separately.